### PR TITLE
[circleci] make forge blocking

### PR DESCRIPTION
**Pull request by rustielin**
Make Forge blocking. Introduce two switches that operators can use to keep Forge from blocking devs in case things go wrong:
* `FORGE_BLOCKING`: if set, forge blocks PRs on failure
* `FORGE_ENABLED`: if set, forge runs on all PRs at land-time, but not necessarily blocking

Also post a better comment to the PR with the actual repro cmd

### CODE REVIEW

**Comment by davidiw** [comment]
Let's get feedback tomorrow on this. I've noticed that forge has been rather flaky, so I'd hate to bring that upon everyone.

**Comment by gregnazario** [comment]
> Let's get feedback tomorrow on this. I've noticed that forge has been rather flaky, so I'd hate to bring that upon everyone.

Yeah, I'd agree for now, let's keep this off to not bring extra burden

File: .circleci/config.yml [hunk 1]
```diff
@@ -142,6 +142,13 @@ jobs:
   forge-k8s-test:
     executor: ubuntu-medium
     steps:
+      - run:
+          name: Check Forge killswitch
+          shell: /bin/bash
+          command: |
+            if [ -n "$FORGE_KILLSWITCH" ]; then
+              circleci-agent step halt
+            fi
```
Comment (gregnazario) [inline .circleci/config.yml:151]: Let's flip this to FORGE_ENABLED.  It's a little clearer than KILLSWITCH being enabled.

File: .circleci/config.yml [hunk 2]
```diff
@@ -169,15 +176,24 @@ jobs:
...
             -X POST -d "{\"body\": \"${FORGE_COMMENT}\"}" \
             "https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/issues/${PR_NUMBER}/comments"
-            exit 0
+      - run:
+          name: Check Forge status
+          shell: /bin/bash
+          command: |
+            if [ -n "$FORGE_NONBLOCKING" ]; then
+              exit 0
+            fi
```
Comment (gregnazario) [inline .circleci/config.yml:195]: Similarly, `FORGE_BLOCKING_ENABLED`, so we can default to false for right now.

File: terraform/testnet/forge/templates/cron.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ metadata:
     {{- include "forge.labels" . | nindent 4 }}
 spec:
   schedule: "0 0 * * *"
+  backoffLimit: 1
```
Comment (gregnazario) [inline terraform/testnet/forge/templates/cron.yaml:9]: Does this make sure it tries again exactly once?

**Review by gregnazario — COMMENTED** [review]
Looks good, we just have to clean it up so we can run non-blocking for now while we collect data.

File: terraform/testnet/forge/templates/cron.yaml (hunk 3 above)
Comment (rustielin) [reply terraform/testnet/forge/templates/cron.yaml]: yea. removed this actually. the default is 6. this is the job that cleans up old forge jobs

File: .circleci/config.yml (hunk 1 above)
Comment (rustielin) [reply .circleci/config.yml]: Done

**Review by sherry-x — APPROVED** [review]

**Pull request merged**